docs/reference/snapshot-restore/restore-snapshot.asciidoc

```diff
@@ -23,6 +23,8 @@ errors>>.
 [[restore-snapshot-prereqs]]
 === Prerequisites
 
+include::register-repository.asciidoc[tag=kib-snapshot-prereqs]
+
 include::apis/restore-snapshot-api.asciidoc[tag=restore-prereqs]
```

Comment:
Once again I can't find the source for this list, but it looks like it contains this item:

> If you restore a data stream, ensure the cluster contains a matching index template with data stream enabled. Without a matching index template, a data stream can’t roll over or create backing indices.

Is this true if you are restoring global state with your snapshot?

Reply:
Thanks for pointing this out. This is intended to cover both cases (existing or restore from cluster state), but it's not clear. I've reworded to:

> Before you restore a data stream, use the [get index template API][0] to ensure the cluster contains a [matching index template][1] with data stream enabled. If no such template exists, you can [create one][1] or [restore a cluster state][2] that contains one. Without a matching index template, a data stream can’t roll over or create backing indices.

[0]: https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-get-template.html
[1]: https://www.elastic.co/guide/en/elasticsearch/reference/master/set-up-a-data-stream.html#create-index-template
[2]: https://www.elastic.co/guide/en/elasticsearch/reference/master/restore-snapshot-api.html#restore-snapshot-api-include-global-state